depends/packages/qrencode.mk

```diff
@@ -1,15 +1,16 @@
 package=qrencode
-$(package)_version=3.4.4
+$(package)_version=4.1.1
```

Comment:
nit 27312: missing update in `doc/dependencies.md`
consider backporting `bitcoin/bitcoin#27441`